### ISSUE TITLE
Mark dataclass and attrs support as experimental

### ISSUE THREAD

**Issue by Gallaecio** [scrapy/itemadapter#32]
Suggested by @kmike during an internal meeting.

If we do this, we should also open a ticket to do something similar in Scrapy.

**Comment by kmike** [comment]
Hm, I was thinking about making their support experimental in itemloaders, not in itemadapter

**Comment by elacuesta** [comment]
I agree with @kmike, dataclass objects do work for all intended purposes within `itemadapter`. The problem in https://github.com/scrapy/itemloaders/issues/14 is about creating objects without all fields being present, but `ItemAdapter` doesn't create items internally.

**Comment by Gallaecio** [comment]
Makes sense.
